language/vm/src/file_format.rs

```diff
@@ -802,7 +803,7 @@ impl CodeUnit {
 ///
 /// Bytecodes operate on a stack machine and each bytecode has side effect on the stack and the
 /// instruction stream.
-#[derive(Clone, Hash, Eq, NumVariants, PartialEq)]
+#[derive(Clone, Hash, Eq, IRBytecode, NumVariants, PartialEq)]
```

Comment:
Same issue here, feels very weird that this exists in the VM

Reply:
While it's a really neat trick to be able to do this, it feels a bit weird to me. 
When new to a compiler project, the way I get clarity on whats going on in a certain pass is to go look at that Pass's IR/AST. But with this macro, I can't really do that. I need to first fully understand the file_format then fully understand the macro that creates this enum. In short, if this were some sort of checked in codegen, it would feel a lot more palatable to me. But as it is now, I don't feel like gain fo deriving this is worth the negative cost from the obfuscation. 

Another consideration, is the code for this generation is relatively straightforward. But if more complicated things are introduced to the file format, it might be hard to update the derivation macro. 

The main question I want to answer is: Is the macro better than just the handwritten enum?